components/match2/wikis/overwatch/match_group_input_custom.lua

```diff
@@ -0,0 +1,525 @@
+---
+-- @Liquipedia
+-- wiki=overwatch
+-- page=Module:MatchGroup/Input/Custom
+--
+-- Please see https://github.com/Liquipedia/Lua-Modules to contribute
+--
+
+local Json = require('Module:Json')
+local Logic = require('Module:Logic')
+local Lua = require('Module:Lua')
+local String = require('Module:StringUtils')
+local Table = require('Module:Table')
+local Variables = require('Module:Variables')
+
+local MatchGroupInput = Lua.import('Module:MatchGroup/Input', {requireDevIfEnabled = true})
+
+local _ALLOWED_STATUSES = { 'W', 'FF', 'DQ', 'L', 'D' }
+local _FINISHED_INDICATORS = { 'skip', 'np', 'cancelled', 'canceled' }
+local _MAX_NUM_OPPONENTS = 8
+local _MAX_NUM_PLAYERS = 10
+local _MAX_NUM_VODGAMES = 9
+local _MAX_NUM_MAPS = 9
+local _DEFAULT_BESTOF = 3
+local _NO_SCORE = -99
+
+local _EPOCH_TIME = '1970-01-01 00:00:00'
+
+-- containers for process helper functions
+local matchFunctions = {}
+local mapFunctions = {}
+local opponentFunctions = {}
+
+local CustomMatchGroupInput = {}
+
+-- called from Module:MatchGroup
+function CustomMatchGroupInput.processMatch(_, match)
+	-- Count number of maps, and automatically count score
+	match = matchFunctions.getBestOf(match)
+	match = matchFunctions.getScoreFromMapWinners(match)
+
+	-- process match
+	Table.mergeInto(
+		match,
+		matchFunctions.readDate(match)
+	)
+	match = matchFunctions.getOpponents(match)
+	match = matchFunctions.getTournamentVars(match)
+	match = matchFunctions.getVodStuff(match)
+	match = matchFunctions.getExtraData(match)
+
+	return match
+end
+
+-- called from Module:Match/Subobjects
+function CustomMatchGroupInput.processMap(_, map)
+	map = mapFunctions.getExtraData(map)
+	map = mapFunctions.getScoresAndWinner(map)
+	map = mapFunctions.getTournamentVars(map)
+
+	return map
+end
+
+-- called from Module:Match/Subobjects
+function CustomMatchGroupInput.processOpponent(_, opponent)
+	-- check for empty opponent and convert to literal
+	if type(opponent) == 'table' and opponent.type == 'team' and Logic.isEmpty(opponent.template) then
+		opponent.name = ''
+		opponent.type = 'literal'
+	end
+
+	-- check for lazy bye's and convert them to literals
+	if type(opponent) == 'table' and string.lower(opponent.template or '') == 'bye' then
+			opponent.name = 'BYE'
+			opponent.type = 'literal'
+	end
+
+	return opponent
+end
```

Comment:
#763

Reply:
I believe i got all the updates from that PR. Let me know if i missed any